src/mixer/basetrackplayer.cpp

```diff
@@ -626,7 +627,36 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
 
     // Update the PlayerInfo class that is used in EngineBroadcast to replace
     // the metadata of a stream
+    QString trackInfo = pNewTrack->getInfo();
+    QString trackInfoArtist = pNewTrack->getArtist();
+    QString trackInfoTitle = pNewTrack->getTitle();
+    QString trackInfoDeck = getGroup();
+    trackInfoDeck.replace("[Channel", "");
+    trackInfoDeck.replace("]", "");
+    // how to call m_settingsPath;
+    QString DeckStatusFilePath = "c:/Users/Evelynne/AppData/Local/Mixxx/controllers/";
+    QString DeckStatusFileLocation = DeckStatusFilePath + "Status" + getGroup() + ".js";
+    
     PlayerInfo::instance().setTrackInfo(getGroup(), m_pLoadedTrack);
+//    QMessageBox::information(nullptr /*or parent*/, "Title", QString("This is an information %1 - %2").arg(getGroup()).arg(trackInfo));
+//    QMessageBox::information(nullptr /*or parent*/, "Title", QString("This is an information %1 - %2").arg(getGroup()).arg(trackInfoPath)); //
+    //  Different file for each Deck / Sampler
+        //QMessageBox::information(nullptr /*or parent*/, "Title", QString("EveStatusFileLocation2: %1").arg(EveStatusFileLocation2)); //
+    QString DeckStatusTxtLine1 = "var TrackDeck" + trackInfoDeck + " = { ";
+    QString DeckStatusTxtLine2 = "Artist : \"" + trackInfoArtist + "\",";
+    QString DeckStatusTxtLine3 = "Title : \"" + trackInfoTitle + "\",";
+    QString DeckStatusTxtLine4 = "};";
+       
+    QFile DeckStatusFile(DeckStatusFileLocation);
+    DeckStatusFile.remove();
+    DeckStatusFile.open(QIODevice::ReadWrite | QIODevice::Append);
```

Comment:
No need for `QIODevice::Append` IMO if you just removed the file, i.e. are creating a new file here.

In general, I think you're better off collecting the data stream first, then write in one step.
Example is writing effects.xml in `EffectsManager::saveEffectsXml()`.

Also, it would be good to clean up, for example when ejecting a track (clear the file, I think this is done somewhere above when loading a null track) and when the player is deleted (remove the file, BaseTrackPlayerImpl::~BaseTrackPlayerImpl). The latter would require to define the filepath in the constructor.

Reply:
Some explication:
-> The 4 "Status[Channelx].js" files are linked in the controller xml, when they are not found at startup Mixxx won't link them when they are filled.
            <file filename="Denon DN-HC4500 4Decks-Scrolling (20240316) H.js" functionprefix=""/>
                -> Header file + declaring the vars and tables to fill with the statusfiles
            <file filename="Status[Channel1].js" functionprefix=""/>
            <file filename="Status[Channel2].js" functionprefix=""/>
            <file filename="Status[Channel3].js" functionprefix=""/>
            <file filename="Status[Channel4].js" functionprefix=""/>
            <file filename="Denon DN-HC4500 4Decks-Scrolling (20240316).midi.js" functionprefix="DenonDNHC4500"/>
- the line  DeckStatusTxtLine4 = "Time : \"" + DeckStatusTime + "\",";
-> When a deck changes the "Status[Channelx].js" Mixxx reloads the whole script, with 4 decks this means that when deck 3/4 are loaded the controller jumps back to 1/2. In my javascript I check the time in the "Status[Channelx].js".
When time in "Status[Channel1].js" < time in "Status[Channel3].js" this means Deck 3 is last manipulatef (load/eject) so the controller has to 'toggle' the deck to control that deck. Else ejecting/loading a track is frustrating.

-> So removing the files is no good, the files have to stay, especialy for the time in it.

Reply:
I have problems putting code here.... lines starting with "<" can't be displayed

Reply:
just use code formatting: 3 x ` at new line before and after code block

Reply:
this is not the right repo, I don't know how this occured... it should be "Nowplaying+DeckStatus+AutoDJCenterXfader " https://github.com/mixxxdj/mixxx/commit/255cb774d9152f2a6f965b17c27294c1220fa0be